src/xmidtsend_rbus.c

```diff
@@ -327,14 +359,70 @@ void sendXmidtEventToServer(wrp_msg_t * msg)
 		if(msg_len > 0)
 		{
 			ParodusInfo("sendUpstreamMsgToServer\n");
-			sendUpstreamMsgToServer(&msg_bytes, msg_len);
+			sendRetStatus = sendUpstreamMsgToServer(&msg_bytes, msg_len);
 		}
+		else
+		{
+			ParodusInfo("The msg_len is zero\n");
+			xmidtQDequeue();
+			wrp_free_struct(notif_wrp_msg);
+			ParodusInfo("After wrp_free_struct\n");
+			free(msg_bytes);
+			msg_bytes = NULL;
+			return;
+		}
+
+		while(sendRetStatus)     //If SendMessage is failed condition
+		{
+			ParodusInfo("sendXmidtEventToServer is Failed\n");
+			if(highQosValueCheck(qos))
+			{
+				errorMsg = strdup("sendXmidtEventToServer Failed , Enqueue event since QOS is High");
```

Comment:
This is not required as we are not sending client disconnect callback now.

Reply:
Removed the change